src/signal.rs

```diff
@@ -482,7 +482,9 @@ pub trait Signal {
     ///     assert_eq!(signal.next(), [4]);
     /// }
     /// ```
-    fn by_ref(&mut self) -> &mut Self {
+    fn by_ref(&mut self) -> &mut Self
+        where Self: Sized
```

Comment:
I'm still unsure why this is necessary? I'm happy to add it if it is necesary, I'm just curious what cause for it is - is there some compile error or warning if we don't have this?

Reply:
Yeah, it means that `Signal` is not object-safe because there's a method that uses `Self` in its return type. This disallows that. The advantage to object safety is that was can return abstract `Box<Signal>` from a function.

Reply:
Ahhh I see, I don't often use trait objects so haven't run into this for a while, but that makes sense!